engine/src/main/java/org/destinationsol/game/asteroid/AsteroidBuilder.java

```diff
@@ -32,20 +33,48 @@
 import org.destinationsol.game.drawables.Drawable;
 import org.destinationsol.game.drawables.DrawableLevel;
 import org.destinationsol.game.drawables.RectSprite;
+import org.json.JSONObject;
+import org.terasology.assets.ResourceUrn;
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Set;
 
 public class AsteroidBuilder {
     private static final float DENSITY = 10f;
     private static final float MAX_A_ROT_SPD = .5f;
     private static final float MAX_BALL_SZ = .2f;
     private final CollisionMeshLoader collisionMeshLoader;
     private final List<TextureAtlas.AtlasRegion> textures;
+    private final String moduleName;
 
-    public AsteroidBuilder() {
-        collisionMeshLoader = new CollisionMeshLoader("engine:asteroids");
-        textures = Assets.listTexturesMatching("engine:asteroid_.*");
+    public AsteroidBuilder(String shipName) {
```

Comment:
Or shipName?

Reply:
This could potentially be changed, but currently it needs a shipName because to find the module the user has selected, it uses the users ship and iterates through all modules to find which module contains that ship.